Library/Homebrew/global.rb

```diff
@@ -26,7 +26,7 @@
 end
 RUBY_BIN = RUBY_PATH.dirname
 
-HOMEBREW_USER_AGENT = "Homebrew #{HOMEBREW_VERSION} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
+HOMEBREW_USER_AGENT = "#{ENV["HOMEBREW_USER_AGENT"]} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
```

Comment:
The stuff inside the `(...)` is a header comment so we can choose our own format. Browsers don't seem to use the slash inside it.

```
# Firefox 45.0.1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:45.0) Gecko/20100101 Firefox/45.0
# Safari 9
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/601.5.17 (KHTML, like Gecko) Version/9.1 Safari/537.86.5
# GNU wget 1.17.1
User-Agent: Wget/1.17.1 (darwin13.4.0)
```

Though since the comments are part of the header, maybe the `(...)` stuff should go inside `ENV["HOMEBREW_USER_AGENT"]` and not be added just inside the Ruby `HOMEBREW_USER_AGENT`. Especially because if we added a sub-product version on (e.g. which version of `curl` we're using), that should go after the `(...)`.